src/libs/actions/App.js

```diff
@@ -128,6 +121,26 @@ function getPolicyIDList(policies) {
 function openApp(policies) {
     API.read('OpenApp', {
         policyIDList: getPolicyIDList(policies),
```

Comment:
For `getPolicyIDList()` I think it is overkill. I think what should really happen is that the `Onyx.connect()` method at the top of `App.js` should just keep an array of policyIDs, as opposed to storing the entire policies and then doing some kind of map to pull out the policyIDs later. Example:

```
const allPolicyIDs = [];
Onyx.connect({
    key: ONYXKEYS.COLLECTION.POLICY,
    callback: (val, key) => {
        if (!val || !key) {
            return;
        }

        allPolicyIDs.push(key);
    },
});
```

If you didn't want to do that, the logic could simply be `_.keys(allPolicies).join(',')` which I think is overkill to have that little bit of logic in a method.

Reply:
Another question, while this is being cleaned up... Why is `policyIDList` a comma separated string instead of just an array of IDs? It seems more simple to just use an array.

Reply:
We can clean up, but the logic is not that simple.

1. Pushing to an array would result in duplicating policyIDs for each call to `Onyx.connect`. I tested this and it seems like each call to ReconnectApp would add another set of policyIDs. Filtering the array or using a Set would solve this though.
2. The keys for the policy object are actually in the `policy_<policyID>` format, so we need to either split it or grab it from the object values. That would get longer than a one liner, which justified this method to me.

Reply:
OK, those are good points. I would revise my solution slightly then to this:

```
const allPolicyIDs = {};
Onyx.connect({
    key: ONYXKEYS.COLLECTION.POLICY,
    callback: (val, key) => {
        if (!val || !key) {
            return;
        }

        allPolicyIDs[key] = key;
    },
});
```

then
```
policyIDList: _.values(allPolicyIDs), // or _.keys() works too. I assume values() is faster though
```

Reply:
haha, I edited it a few times, but I think I like that version ^

Reply:
I edited this a bit in case we end up going with the `waitForCollectionCallback` param.